packages/braintree-integration/src/braintree-local-payment-methods/braintree-local-methods-payment-strategy.ts

```diff
@@ -0,0 +1,237 @@
+        }
+    }
+
+    private renderButton(methodId: string) {
+        if (this.braintreeLocalMethods) {
+            let buttonContainer;
+            const { buttonText, container, onRenderButton } = this.braintreeLocalMethods;
+            const buttonContainerId = container.split('#')[1];
+            const lpmButton = document.createElement('button');
+            buttonContainer = document.getElementById(buttonContainerId);
+            const className = buttonContainer?.getAttribute('class');
+            lpmButton.setAttribute('class', className || '');
+            lpmButton.setAttribute('id', methodId);
+            lpmButton.innerText = buttonText;
+            lpmButton.addEventListener('click', (e) => this.handleClick(e, methodId));
+
+            if (onRenderButton && typeof onRenderButton === 'function') {
+                onRenderButton();
+            }
+
+            buttonContainer = document.getElementById(buttonContainerId);
+
+            if (buttonContainer) {
+                buttonContainer.append(lpmButton);
+            }
+        }
+    }
+
+    /**
+     *
+     * Loading Indicator methods
+     *
+     * */
+    private toggleLoadingIndicator(isLoading: boolean): void {
+        if (isLoading && this.loadingIndicatorContainer) {
+            this.loadingIndicator.show(this.loadingIndicatorContainer);
+        } else {
+            this.loadingIndicator.hide();
+        }
+    }
+
+    private handleClick(e: Event, methodId: string) {
+        e.preventDefault();
+        this.toggleLoadingIndicator(true);
+        const state = this.paymentIntegrationService.getState();
+        const cart = state.getCartOrThrow();
+        const billing = state.getBillingAddressOrThrow();
+        const { firstName, lastName, countryCode } = billing;
+        const { baseAmount, currency, email } = cart;
+        const isShippingRequired = cart.lineItems.physicalItems.length > 0;
+        const { submitForm } = this.braintreeLocalMethods || {};
+
+        this.localPaymentInstance?.startPayment(
+            {
+                paymentType: methodId,
+                amount: baseAmount,
+                fallback: {
+                    url: '',
+                    buttonText: '',
+                },
+                currencyCode: currency.code,
+                shippingAddressRequired: isShippingRequired,
+                email,
+                givenName: firstName,
+                surname: lastName,
+                address: {
+                    countryCode,
+                },
+                onPaymentStart: (data: onPaymentStartData, start: () => void) => {
+                    // Call start to initiate the popup
+                    this.orderId = data.paymentId;
+                    start();
+                },
+            },
+            (startPaymentError: StartPaymentError, payload: LocalPaymentsPayload) => {
+                if (startPaymentError.code !== 'LOCAL_PAYMENT_WINDOW_CLOSED') {
```

Comment:
What should we do in cases when there will be an error with another `startPaymentError.code`

Reply:
If we have this code we don't need to handle error because it is indicates that we closed iframe but in other cases it handles error whatever we recieved

Reply:
Ah, my apologies. Everything ok, just double checked the condition